ecs-init/config/common.go

```diff
@@ -117,11 +122,14 @@ func DockerUnixSocket() (string, bool) {
 
 // getBaseLocationForRegion fetches the bucket URI from list of S3 Buckets by region name or default if key is not found
 func getBaseLocationForRegion(regionName string) string {
+	if regionName == RegionNameNotFound {
```

Comment:
There are 3 possible cases that are handled here:

1. regionName == RegionNameNotFound: Couldn't resolve your region for some reason
2. regionName == "us-east-2": Not explicitly in the map
3. regionName == "cn-north-1" (or some other map member): Key is in the map

Your [earlier comments](https://github.com/aws/amazon-ecs-init/pull/87#discussion_r119472709) led @cabbruzzese and me to see that the region resolving code was not separating the case of an unresolved region and a default region - which would have been intentionally passed to this function if desired - and thus we scoped the defaulting/fallback behavior to this function.